core/schema/env.graphqls

```diff
@@ -0,0 +1,252 @@
+extend type Query {
+  "The environment initialized from the given ID."
+  environment(id: EnvironmentID): Environment!
+
+  """
+  Install the given environment into this graphql API. Its schema will be
+  stitched into the schema of this server, making those APIs available for
+  subsequent queries.
+
+  If an environment with the same ID has already been installed, this is a no-op.
+
+  If there are any conflicts between the environment's schema and any existing
+  schemas, an error will be returned.
+  """
+  installEnvironment(id: EnvironmentID!): Boolean!
+
+  "The check initialized from the given ID."
+  check(id: CheckID): Check!
+
+  "The check result initialized from the given ID."
+  checkResult(id: CheckResultID): CheckResult!
+
+  "A check result initialized with the given success and output."
+  staticCheckResult(success: Boolean!, output: String): CheckResult!
+
+  """
+  The environment the requester is being executed in (or an error if none).
+  """
+  currentEnvironment: Environment!
+}
+
+"A unique environment identifier."
+scalar EnvironmentID
+
+"A unique environment check identifier."
+scalar CheckID
+
+"A unique environment check result identifier."
+scalar CheckResultID
+
+"A group of Dagger entrypoints that can be queried and/or invoked."
+type Environment {
+  "A unique identifier for this environment."
+  id: EnvironmentID!
+
+  "Name of the environment"
+  name: String
+
+  "The directory containing all the source needed to execute this environment's code"
+  sourceDirectory: Directory!
+
+  "The SDK that this environment will be executed with"
+  sdk: String!
+
+  """
+  Other environments that this environment depends on. If this environment is installed,
+  all dependencies will be (recursively) installed too.
+  """
+  dependencies: [Environment!]
+
+  """
+  The environment initialized with the given name, sourceDirectory, sdk and dependencies.
+
+  If set, sourceDirectorySubpath should point to the subpath of sourceDirectory that
+  contains the environment code. If unset, it will default to the root of the sourceDirectory.
+  """
+  from(
```

Comment:
nit: `Container.From` gets a pass from me because it's so common and maps well to Dockerfile patterns, but this `from` feels like cheating. :joy: 

On the other hand, there's no real precedent for this. Everything else has a valid "zero" state which lets us build everything incrementally like `withName` and `withSDK`. Maybe this is just the easiest default name when there are multiple required values? Seems like the lesser evil to partial instantiation. The alternatives would be `new` or `create` but we don't want it to sound imperative, or that you are creating multiple instances when given the same args. Hmm...

Reply:
Haha yeah you pretty much outlined the entire saga I went through here.

I started out trying to refactor the env schema to use `with*` on everything, but quickly realized it gets extremely confusing (both for users and us) that gets.

So I then pivoted to only have `with*` on the fields which make sense to change after initialization.

And once I reached that point, my brain was consumed 100% with refactoring code, so I went with `from` because I didn't have time/neurons to think about it more 😄

But totally agree this name is worth re-thinking. It makes sense on some level in that you are creating an environment *from* those parameters, but I don't know how obvious that is.

... I'm kind of drawing a blank here still 🤔. I'll keep thinking though.